runtime/terminator.h

```diff
@@ -0,0 +1,42 @@
+//===-- runtime/terminator.h ------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//----------------------------------------------------------------------------//
+
+// Termination of the image
+
+#ifndef FORTRAN_RUNTIME_TERMINATOR_H_
+#define FORTRAN_RUNTIME_TERMINATOR_H_
+
+#include "entry-names.h"
+#include <cstdarg>
+
+namespace Fortran::runtime {
+
+// A mixin class for statement-specific image error termination
+// for errors detected in the runtime library
+class Terminator {
```

Comment:
Who will own/create terminator objects, the runtime or the user program ? For I/O for instance, will it be stored in the `Cookie` in an opaque way to the user program ?

Reply:
This is just for the runtime.  Instances will be part of per-I/O-statement state.